src/core/server/plugins/discovery/plugins_discovery.ts

```diff
@@ -115,11 +144,13 @@ function processPluginSearchPaths$(pluginDirs: ReadonlyArray<string>, log: Logge
  */
 function createPlugin$(path: string, log: Logger, coreContext: CoreContext) {
   return from(parseManifest(path, coreContext.env.packageInfo)).pipe(
-    map(manifest => {
+    mergeMap(async manifest => {
       log.debug(`Successfully discovered plugin "${manifest.id}" at "${path}"`);
+
       return new PluginWrapper(
         path,
         manifest,
+        readSchemaMaybe(path, manifest, log),
```

Comment:
__question:__ have you considered making this a part of the `PluginWrapper` instead? It already "knows" how to "require" plugin so we can share this knowledge and don't spread it over multiple places/files. We could potentially have a `PluginWrapper.getConfigSchema()` that is called only when it's needed.